rasterio/_io.pyx

```diff
@@ -1590,6 +1595,54 @@ cdef class DatasetWriterBase(DatasetReaderBase):
                 if factors_c != NULL:
                     CPLFree(factors_c)
 
+    def set_gcps(self, gcps, crs=None):
+        cdef char *srcwkt = NULL
+        cdef GDAL_GCP *gcplist = <GDAL_GCP *>CPLMalloc(len(gcps) * sizeof(GDAL_GCP))
+
+        try:
+            for i, obj in enumerate(gcps):
+                ident = str(i).encode('utf-8')
+                info = "".encode('utf-8')
```

Comment:
I'd imagine users would want to retain the id/info from the original gcps. It's valuable metadata if you're working directly with the gcps.